source/extensions/matching/network/common/inputs.h

```diff
@@ -66,13 +66,12 @@ DECLARE_FACTORY(HttpDestinationIPInputFactory);
 template <class MatchingDataType>
 class DestinationPortInput : public Matcher::DataInput<MatchingDataType> {
 public:
-  Matcher::DataInputGetResult get(const MatchingDataType& data) const override {
+  Matcher::DataInputGetResult<std::string> get(const MatchingDataType& data) const override {
     const auto& address = data.localAddress();
     if (address.type() != Network::Address::Type::Ip) {
-      return {Matcher::DataInputGetResult::DataAvailability::AllDataAvailable, absl::nullopt};
+      return {Matcher::DataAvailability::AllDataAvailable, absl::nullopt};
     }
-    return {Matcher::DataInputGetResult::DataAvailability::AllDataAvailable,
-            absl::StrCat(address.ip()->port())};
+    return {Matcher::DataAvailability::AllDataAvailable, absl::StrCat(address.ip()->port())};
```

Comment:
This should really be an integer input value, not a string. Can we try changing it and see how matchers adapt to it?

Reply:
Yes, but as i mentioned in the PR description, I plan to do it in following PRs as this essentially requires second part of changes --- make input matcher interface generic. 

So the reason I don't change it here now is 1) to avoid making the PR too large to review and 2) to fit the timeline of two import projects that depends on this API with CEL. I plan to do INT case after CEL use case as CEL should be able provide good idea of how this API is being used.

I fully agree that it is better to give the reviewers whole picture and more details , so I will write up a doc for review and discussion.